numcodecs/tests/test_entrypoints_backport.py

```diff
@@ -0,0 +1,32 @@
+import os.path
+import pkgutil
+import sys
+
+import pytest
+
+from multiprocessing import Process
+
+import numcodecs.registry
+
+if not pkgutil.find_loader("importlib_metadata"):
+    pytest.skip("This test module requires importlib_metadata to be installed")
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+def get_entrypoints_with_importlib_metadata_loaded():
+    # importlib_metadata patches importlib.metadata, which can lead to breaking changes
+    # to the APIs of EntryPoint objects used when registering entrypoints. Attempt to
+    # isolate those changes to just this test.
+    import importlib_metadata  # noqa: F401
+    sys.path.append(here)
+    numcodecs.registry.run_entrypoints()
+    cls = numcodecs.registry.get_codec({"id": "test"})
+    assert cls.codec_id == "test"
```

Comment:
Would it make sense to reuse `set_path` here somehow? Should cleanup `sys.path` and the `numcodecs.registry` after?

Reply:
I thought about reusing, but I think it'd be more work than it's worth. We can't use a fixture directly because it's not supposed to affect the actual test function, we need it to affect the helper function that's run in the subprocess and that function isn't a test so pytest won't patch in fixtures. We could try and split it into a helper function like I had initially, but it's not really necessary. Since the function is running in a subprocess we don't need to clean things up, it shouldn't affect the modules in the parent process.